File: proposed-projects/round-trip-experiment/reverse/20260909T132655Z_2a16d72b/20260910T075017Z_244b957c/ported_implementation/src/utils/errors/get-input-as-string.ts
export type ValidInput = string | number | number[];

export const getInputAsString = (src: ValidInput): string => {
  if (typeof src === 'string') {
    return src;
  }
  const codePoints = Array.isArray(src) ? src : [src];
  return codePoints.map((cp) => String.fromCodePoint(cp)).join('');
};

File: packages/gbnf-experiment/data/20260909T193332Z_f85fdfbb/ported_implementation/utils/errors/get_input_as_string.py
from __future__ import annotations

from typing import TYPE_CHECKING

from ..utf16 import to_utf16_units

if TYPE_CHECKING:  # pragma: no cover
    from ...grammar_graph.types import ValidInput


def get_input_as_string(src: "ValidInput") -> str:
    if isinstance(src, str):
        return to_utf16_units(src)
    code_points = list(src) if isinstance(src, (list, tuple)) else [src]
    return "".join(chr(cp) for cp in code_points)
